### ISSUE TITLE
messagePanel not loading, TypeError: this.context.matrixClient.getRoom(...) is null

### ISSUE THREAD

**Issue by grahamperrin** [vector-im/element-web#5893]
### Description

A first click on the bell results in an asynchronous progress indicator spinning apparently endlessly. 

Click the head and shoulders then again the bell, there remains the list of people (neither notifications, nor a progress indicator). 

This problem may be specific to `/develop`/

### Steps to reproduce

- Firefox
- new profile
- https://riot.im/develop/#/room/#riot:matrix.org
- sign in
- open the browser console
- in Riot, click the bell

#### Expected

- notifications (maybe preceded by a brief spin of an asynchronous progress indicator)

#### Actual results

- asynchronous progress indicator, spinning apparently endlessly, no notifications

```text
TypeError: this.context.matrixClient.getRoom(...) is null
```
![2017-12-28 14 54 14](https://user-images.githubusercontent.com/192271/34414409-1ffb3b3e-ebe1-11e7-8dbd-23a722632ed2.png)

![2017-12-28 14 55 14](https://user-images.githubusercontent.com/192271/34414422-3597b170-ebe1-11e7-89a8-dbf1da6adad3.png)

Copied from source: 

```text
	        var relatedGroupsEvent = this.context.matrixClient.getRoom(this.props.mxEvent.getRoomId()).currentState.getStateEvents('m.room.related_groups', '');
```
- click away from the bell (e.g. head and shoulders)then again on the bell
- result: 

```text
can't initialise scroll state because messagePanel didn't load
```

----

I might have sent a bug report, with a log, yesterday, whilst using a non-clean browser profile. 

I'll send one again, with a clean profile, with reference to this bug report. 

### Version information

Firefox 56.0.2 (64-bit)

FreeBSD-CURRENT

```text
[grahamperrin@momh167-gjp4-hpelitebook8570p-freebsd] ~% uname -v
FreeBSD 12.0-CURRENT #0 r320869: Mon Jul 10 13:57:55 UTC 2017     root@releng3.nyi.freebsd.org:/usr/obj/usr/src/sys/GENERIC 
```
Homeserver is https://matrix.org
Identity Server is https://vector.im
matrix-react-sdk version: <local>
riot-web version: cf47b919d547-react-b5f6d97fa2fc-js-f2b7e8b038bc
olm version: 2.2.1

**Comment by grahamperrin** [comment]
> … I'll send (a log) again, with a clean profile, with reference to this bug report. …

Done. 

> … This problem may be specific to `/develop/` … 

That seems to be true. GUI symptoms of the issue are currently: 

- not reproducible at https://riot.im/app/#/room/#riot:matrix.org in the environment outlined above
- not reproducible at https://riot.im/app/#/room/#riot:matrix.org in Waterfox 56.0.1 on the same machine
- **reproducible** at https://riot.im/develop/#/room/#riot:matrix.org in Waterfox 56.0.1 on the same machine
- **reproducible** at https://riot.im/develop/#/room/#riot:matrix.org in Chromium 61.0.3163.100 (Developer Build) (64-bit) on the same machine.

**Comment by lukebarnard1** [comment]
It looks like you've received a notification for an event in a room that your client does not know about. Most components (including `SenderProfile`, the offending component here) assume that the client has a room associated with the event it uses.

**Comment by grahamperrin** [comment]
> … null-guard on the result of `getRoom(...)` …

Sounds complicated :-) I might need a walk-through. No rush, though …

**Comment by grahamperrin** [comment]
I might have stumbled across a workaround: 

1. alongside _HISTORICAL_, click the disclosure triangle
2. in the _Filter room names_ field, type something
3. observe the asynchronous progress indicator spinning for a very long time
4. without waiting for the spin to cease, clear the filter
5. click the bell icon.

Example results of steps (3) and (4), where `#palemoon` was a room that I left earlier this morning (whilst using `/app/`): 

![2018-01-05 03 40 04 riot 4](https://user-images.githubusercontent.com/192271/34596248-185b01a2-f1d6-11e7-9f4e-4bdccb973fc5.png)

![2018-01-05 03 41 39 riot 4](https://user-images.githubusercontent.com/192271/34596245-14407dae-f1d6-11e7-974f-6e2be7f00312.png)

----

Rewind a little, to the `/app/` session … if I recall correctly: 

- the initial disclosure of HISTORICAL presented just the one room
- the other rooms appeared only after steps (2)–(4) above

– @lukebarnard1, do those two points fit with this part of your comment? 

> … a room that your client does not know about. …

**Comment by grahamperrin** [comment]
OK, looking more closely at a screen recording (below) of the bug and a workaround in Chromium, I reckon that the steps above were convoluted. The filtering might be coincidental. 

[2018-01-05 03:50.ogv.zip](https://github.com/vector-im/riot-web/files/1605618/2018-01-05.03.50.ogv.zip)

## tl;dr

The workaround might be, simply **trigger an update of the HISTORICAL list of rooms**, and wait (maybe a minute or two) for the list to appear. Then the bell icon works as expected.

**Comment by lukebarnard1** [comment]
Loading historical rooms will give your client the room it needs and prevent the NPE. This really just needs a null-guard and is far from complicated.

**Comment by t3chguy** [comment]
Closing due to it likely being fixed
